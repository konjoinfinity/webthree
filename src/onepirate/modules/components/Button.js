import * as React from 'react';
import { experimentalStyled as styled } from '@mui/material/styles';
import MuiButton from '@mui/material/Button';

const ButtonRoot = styled(MuiButton)(({ theme, size }) => ({
  borderRadius: 0,
  fontWeight: theme.typography.fontWeightMedium,
  fontFamily: theme.typography.h1.fontFamily,
  padding: theme.spacing(2, 4),
  fontSize: theme.typography.pxToRem(14),
  boxShadow: 'none',
  '&:active, &:focus': {
    boxShadow: 'none',
  },
  ...(size === 'small' && {
    padding: theme.spacing(1, 3),
    fontSize: theme.typography.pxToRem(13),
  }),
  ...(size === 'large' && {
    padding: theme.spacing(2, 5),
    fontSize: theme.typography.pxToRem(16),
  }),
}));

function Button(props) {
  return <ButtonRoot {...props} />;
}

export default Button;
