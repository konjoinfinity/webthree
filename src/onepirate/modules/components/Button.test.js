import React from "react"
import { render, fireEvent } from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"
import Button from "./Button"

  it("should render as default button", () => {
    const { container } = render(<Button />)
    expect(container.firstChild).toMatchSnapshot()
  })

it('should render a large button', () => {
  const { container } = render(<Button size="large"/>)
  expect(container).toMatchSnapshot()
  expect(container.firstChild).toBeInTheDocument();
  expect(container.firstChild).toHaveClass("MuiButton-sizeLarge")
})

it('should render a small button', () => {
  const { container } = render(<Button size="small"/>)
  expect(container).toMatchSnapshot()
  expect(container.firstChild).toBeInTheDocument();
  expect(container.firstChild).toHaveClass("MuiButton-sizeSmall")
})

it('should render a secondary button', () => {
  const variant = 'secondary'
  const { container } = render(<Button color={variant}></Button>)
  expect(container).toMatchSnapshot()
  expect(container.firstChild).toHaveClass("MuiButton-textSecondary")
})

it('should render the button with some text', () => {
  const text = 'Click me'
  const { getByText, container } = render(<Button>{text}</Button>)
  expect(container).toMatchSnapshot()
  getByText(text)
})

it('should call the click callback handler', () => {
  const content = 'Click Me!'
  const onClick = jest.fn()
  const { getByText } = render(<Button onClick={onClick}>{content}</Button>)
  fireEvent.click(getByText(content))
  expect(onClick).toHaveBeenCalledTimes(1)
})