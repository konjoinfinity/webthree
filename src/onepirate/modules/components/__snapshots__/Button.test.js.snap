// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a large button 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge css-88gtui-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`should render a secondary button 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium css-cmd9fq-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`should render a small button 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-qntokj-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`should render as default button 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-11bier9-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`should render the button with some text 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-11bier9-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Click me
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
